tests/testthat/test-drive_trash.R

```diff
@@ -0,0 +1,52 @@
+context("Trash files")
+
+## WARNING: this will empty your drive trash. If you do
+## not want that to happen, set EMPTY_TRASH = FALSE
+# EMPTY_TRASH = TRUE
+
+## NOTE if you do not currently have the files needed,
+## change run & clean below to TRUE to create files needed
+## (CAUTION, this will delete files that will interfere)
+
+
+nm_ <- nm_fun("-TEST-drive-trash")
+
+run <- FALSE
+clean <- FALSE
+if (run) {
+  ## make sure directory is clean
+  if (clean) {
+    del <- drive_delete(nm_("foo"), verbose = FALSE)
+  }
+  drive_mkdir(nm_("foo"))
+}
+
+test_that("drive_trash() moves object to the trash", {
+
+  skip_on_travis()
+  skip_on_appveyor()
```

Comment:
Maybe this and test below should be merged, so they can work together, actually.